### PR TITLE
Feature/field registry

### PR DESCRIPTION
**Pull request by timofeymukha**
This PR will be use to track the developments as per issue #612
I currently have a crude skeleton, which appears to work in the sense that I can run the tgv example.

### CODE REVIEW

**Comment by timofeymukha** [comment]
`  2   src/field/field_registry.f90                       10.00      32`
Top score from the linter? :-)

**Comment by njansson** [comment]
Isn't this PR missing an updated `Makefile.am` ?

File: src/fluid/fluid_method.f90 [hunk 1]
```diff
@@ -354,11 +358,15 @@ subroutine fluid_scheme_init_all(this, msh, lx, params, &
...
-    call field_init(this%p, this%dm_Xh, 'p')
+
+   call neko_field_registry%add_field(this%dm_Xh, 'u')
+   call neko_field_registry%add_field(this%dm_Xh, 'v')
+   call neko_field_registry%add_field(this%dm_Xh, 'w')
+   call neko_field_registry%add_field(this%dm_Xh, 'p')
+   this%u => neko_field_registry%get_field('u')
+   this%v => neko_field_registry%get_field('v')
+   this%w => neko_field_registry%get_field('w')
+   this%p => neko_field_registry%get_field('p')
```
Comment (njansson) [inline src/fluid/fluid_method.f90:369]: Minor formatting things
```suggestion
    call neko_field_registry%add_field(this%dm_Xh, 'u')
    call neko_field_registry%add_field(this%dm_Xh, 'v')
    call neko_field_registry%add_field(this%dm_Xh, 'w')
    call neko_field_registry%add_field(this%dm_Xh, 'p')
    this%u => neko_field_registry%get_field('u')
    this%v => neko_field_registry%get_field('v')
    this%w => neko_field_registry%get_field('w')
    this%p => neko_field_registry%get_field('p')
```

File: src/fluid/fluid_method.f90 [hunk 2]
```diff
@@ -328,9 +329,12 @@ subroutine fluid_scheme_init_uvw(this, msh, lx, params, kspv_init, scheme)
...
     
-    call field_init(this%u, this%dm_Xh, 'u')
-    call field_init(this%v, this%dm_Xh, 'v')
-    call field_init(this%w, this%dm_Xh, 'w')
+   call neko_field_registry%add_field(this%dm_Xh, 'u')
+   call neko_field_registry%add_field(this%dm_Xh, 'v')
+   call neko_field_registry%add_field(this%dm_Xh, 'w')
+   this%u => neko_field_registry%get_field('u')
+   this%v => neko_field_registry%get_field('v')
+   this%w => neko_field_registry%get_field('w')
```
Comment (njansson) [inline src/fluid/fluid_method.f90:337]: Minor formatting things
```suggestion
    call neko_field_registry%add_field(this%dm_Xh, 'u')
    call neko_field_registry%add_field(this%dm_Xh, 'v')
    call neko_field_registry%add_field(this%dm_Xh, 'w')
    this%u => neko_field_registry%get_field('u')
    this%v => neko_field_registry%get_field('v')
    this%w => neko_field_registry%get_field('w')
```

File: src/fluid/fluid_method.f90 [hunk 3]
```diff
@@ -62,14 +62,15 @@ module fluid_method
...
   !> Base type of all fluid formulations
   type, abstract :: fluid_scheme_t
-     type(field_t) :: u         !< x-component of Velocity
-     type(field_t) :: v         !< y-component of Velocity
-     type(field_t) :: w         !< z-component of Velocity
-     type(field_t) :: p         !< Pressure
+     type(field_t), pointer :: u         !< x-component of Velocity
+     type(field_t), pointer :: v         !< y-component of Velocity
+     type(field_t), pointer :: w         !< z-component of Velocity
+     type(field_t), pointer :: p         !< Pressure
```
Comment (njansson) [inline src/fluid/fluid_method.f90:73]: Do we need to nullify these? Or I guess we will never try to free them from `fluid_method`

**Review by njansson — CHANGES_REQUESTED** [review]

**Review by njansson — APPROVED** [review]

**Pull request merged**